lib/upgrade/etcd/conversionv1v3/hostendpoint_test.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conversionv1v3
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	apiv1 "github.com/projectcalico/libcalico-go/lib/apis/v1"
+	"github.com/projectcalico/libcalico-go/lib/apis/v1/unversioned"
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	//"github.com/projectcalico/libcalico-go/lib/ipip"
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"github.com/projectcalico/libcalico-go/lib/numorstring"
+)
+
+const hepName = "my-hep"
+const nodeName = "my-node"
+const profileName = "my-prof"
+const ifName = "eth3"
+const portName = "my-port"
+const portNum = 12345
+const portProtocol = "tcp"
+const ip = "192.168.1.1"
+
+var hepTable = []struct {
+	description string
+	v1API unversioned.Resource
+	v1KVP *model.KVPair
+	v3API apiv3.HostEndpoint
+}{
+	{
+		description: "Super basic hep",
+		v1API: apiv1.HostEndpoint{
+			Metadata: apiv1.HostEndpointMetadata{
+				Name: hepName,
+			},
+			Spec: apiv1.HostEndpointSpec{},
+		},
+		v1KVP: &model.KVPair{
+			Key: model.HostEndpointKey{
+				EndpointID: hepName,
+			},
+			Value: &model.HostEndpoint{
+			},
+		},
+		v3API: apiv3.HostEndpoint{
+			ObjectMeta: v1.ObjectMeta{
+				Name: hepName,
+			},
+			Spec: apiv3.HostEndpointSpec{
+				ExpectedIPs: []string{},
```

Comment:
Is this ok? I'm finding its required because of how I initialize the IPs as an empty array here: https://github.com/projectcalico/libcalico-go/pull/701/files#diff-61a4c1332180370f6ebb2f38bf4d8542R86